ui/src/lib/api.ts

```diff
@@ -1,32 +1,111 @@
 import { Target, Listener, LocalConfig, Bind, Backend, Route } from "./types";
 
+// Mapping utilities are centralized in configMapper
+import { configDumpToLocalConfig } from "./configMapper";
+
 const API_URL = process.env.NODE_ENV === "production" ? "" : "http://localhost:15000";
 
+let xdsMode = false;
+let xdsModeKnown = false;
+
+export function isXdsMode() {
+  return xdsMode;
+}
+
+export function isXdsModeKnown() {
+  return xdsModeKnown;
+}
+
+// If the mode has not yet been determined, makes a request to /config_dump
+// to establish it, broadcasting the result so that subscribers update as well.
+export async function ensureXdsModeLoaded(): Promise<boolean> {
+  if (xdsModeKnown) {
+    return xdsMode;
+  }
+
+  try {
+    const resp = await fetch(`${API_URL}/config_dump`);
+    if (resp.ok) {
+      const dumpJson = await resp.json();
+      const enabled = !!dumpJson?.config?.xds?.address;
+      setAndBroadcastXds(enabled);
+      return enabled;
+    }
+  } catch (err) {
+    console.error("Failed to determine whether XDS mode is enabled", err);
+  }
+
+  // If we could not determine the mode, assume whatever default we currently have
+  return xdsMode;
+}
+
+type XdsSubscriber = (val: boolean) => void;
+const xdsSubscribers: XdsSubscriber[] = [];
+
+function setAndBroadcastXds(val: boolean) {
```

Comment:
nit: consider moving these setter functions and subscribe functions to the use-xds-mode -- seems they be a better fit there